### PR TITLE
Add Naive Bayes DB integration & Begin work on Uploads

### PR DESCRIPTION
**Pull request by priyeshpatel**

### CODE REVIEW

File: src/main/java/uk/ac/cam/cl/retailcategorymapper/db/UploadDb.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+     * List all uploads stored in the database.
+     * @return List of uploads.
+     */
+    public static List<Upload> getUploads() {
+        Jedis jedis = JedisWrapper.getInstance();
+        Redisson redisson = RedissonWrapper.getInstance();
+
+        Set<String> keys = jedis.keys(KeyBuilder.allUploadInstances());
+
+        List<Upload> result = new ArrayList<>();
```
Comment (ivonchyk) [inline src/main/java/uk/ac/cam/cl/retailcategorymapper/db/UploadDb.java]: should this not be new ArrayList<Upload>()?

File: src/main/java/uk/ac/cam/cl/retailcategorymapper/db/NaiveBayesDb.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            Taxonomy taxonomy) {
+        Redisson redisson = RedissonWrapper.getInstance();
+
+        String key = KeyBuilder.naiveCategoryProductCount(taxonomy.getId());
+
+        return redisson.getMap(key);
+    }
+
+    /**
+     * Get a category to feature count map for a taxonomy.
```
Comment (ivonchyk) [inline src/main/java/uk/ac/cam/cl/retailcategorymapper/db/NaiveBayesDb.java:71]: This returns just a category -> Integer, not involving features at all? Just a slightly confusing javadoc

File: src/main/java/uk/ac/cam/cl/retailcategorymapper/db/UploadDb.java (hunk 1 above)
Comment (priyeshpatel) [inline src/main/java/uk/ac/cam/cl/retailcategorymapper/db/UploadDb.java:26]: As in, without the `<>`? 

Java 7 added diamond notation where it detects the type.

File: src/main/java/uk/ac/cam/cl/retailcategorymapper/db/NaiveBayesDb.java (hunk 2 above)
Comment (priyeshpatel) [inline src/main/java/uk/ac/cam/cl/retailcategorymapper/db/NaiveBayesDb.java:71]: This is the total number of features in a category so it's per category. `getCategoryFeatureObservationMap` is per feature.

**Pull request merged**